docs/topics/unused-return-value-checker.md

```diff
@@ -0,0 +1,198 @@
+[//]: # (title: Unused return value checker)
+
+<primary-label ref="experimental-general"/>
+
+> This feature is planned to be stabilized and improved in future Kotlin releases.
+> We would appreciate your feedback in our issue tracker [YouTrack](https://youtrack.jetbrains.com/issue/KT-12719).
+> 
+> For more information, see the related [KEEP proposal](https://github.com/Kotlin/KEEP/blob/main/proposals/KEEP-0412-unused-return-value-checker.md).
+>
+{style="note"}
+
+The unused return value checker allows you to detect _ignored results_.
+These are values returned from expressions that produce something other than
+`Unit`, `Nothing`, or `Nothing?` and aren't:
+
+* Stored in a variable or property.
+* Returned or thrown.
+* Passed as an argument to another function.
+* Used as a receiver in a call or safe call.
+* Checked in a condition such as `if`, `when`, or `while`.
+* Used as the last statement of a lambda.
+
+The checker doesn't report ignored results for increment operations like `++` and `--`,
+or for boolean shortcuts where the right-hand side exits the current function, for example, `condition || return`.
+
+You can use the unused return value checker to catch bugs where a function call produces a meaningful result, but the result is silently dropped.
+This helps prevent unexpected behavior and makes such issues easier to track down.
+
+Here's an example:
+
+```kotlin
+fun formatGreeting(name: String): String {
+    if (name.isBlank()) return "Hello, anonymous user!"
+    if (!name.contains(' ')) {
+        // The checker reports a warning that this result is ignored:
+        // Unused return value of 'plus'.
+        "Hello, " + name.replaceFirstChar(Char::titlecase) + "!"
+    }
+    val (first, last) = name.split(' ')
+    return "Hello, $first! Or should I call you Dr. $last?"
+}
+```
+
+In this example, a string is created but never used, so the checker reports it as an ignored result.
```

Comment:
What about bringing this explanation to the leading line, so that it has context from the beginning. 

Here's an example of a string that is created but never used...